business/mesh.go

```diff
@@ -133,7 +133,14 @@ func (in *MeshService) discoverKiali(ctx context.Context, clusterName string, r
 	for _, d := range services.Items {
 		kiali := convertKialiServiceToInstance(&d)
 		// If URL is already populated (because of an annotation), trust that because it's user configuration.
-		// Only guess ourselves on our own cluster.
+		// But if Kiali URL configured per cluster name, instance name and namespace, then use that URL.
+		for _, cfgurl := range config.Get().KialiFeatureFlags.Clustering.KialiURLs {
+			if cfgurl.ClusterName == clusterName && cfgurl.InstanceName == kiali.ServiceName && cfgurl.Namespace == kiali.Namespace {
+				kiali.Url = cfgurl.URL
+				break
+			}
+		}
```

Comment:
I don't think this enables the main use case for this feature which is: having a url to an external kiali instance _without_ having to give kiali access to the cluster that external kiali instance is running on. If Kiali is only creating a `KialiInstance` for every service that it reads from the external cluster, when Kiali does not have access to the external cluster it will not create a `KialiInstance` for it even though it has the url field set on the config. I think this will require more changes than adding the url here.

Reply:
The feature request said  "_There needs to be a way to for Kiali to generate links to external Kialis without requiring access to external clusters. This can be a simple as adding a config value that Kiali would use to generate these links._ " 
If the Kiali does not have access to remote cluster, but the configuration contains a URL, then it can be shown to user, but will it be OK from UX and security perspective?

Reply:
@nrfox you mean basically show another lines in Mesh with a 'Cluster Name' and 'Kiali' link columns (the rest of columns empty if unable to access) for those clusters which are not accessible but are set in config?

Reply:
> If the Kiali does not have access to remote cluster, but the configuration contains a URL, then it can be shown to user, but will it be OK from UX and security perspective?

Presumably the user would have access even though Kiali doesn't. It's a separate Kiali so the user would have to login to that Kiali separately.

> @nrfox you mean basically show another lines in Mesh with a 'Cluster Name' and 'Kiali' link columns (the rest of columns empty if unable to access) for those clusters which are not accessible but are set in config?

Yes and in addition to the cluster column on the mesh page there are links that appear in the side panel when you click on a node on the graph. These should link to the external kiali if one is set.

Reply:
> 
> Yes and in addition to the cluster column on the mesh page there are links that appear in the side panel when you click on a node on the graph. These should link to the external kiali if one is set.

@nrfox which node's side panel in Graph contains links to Kiali?